### PR TITLE
Switch boost to new bzlmod repos

### PR DESCRIPTION
**Pull request by hofbi**
Boost now provides many modules available on BCR. All boost deps of ROS are meanwhile there so we can switch to this fine granular approach.

### CODE REVIEW

File: ros/repositories/ros_comm.BUILD.bazel [hunk 1]
```diff
@@ -130,11 +130,11 @@ cc_ros_library(
...
         ":xmlrpcpp",
-        "@boost//:chrono",
-        "@boost//:filesystem",
-        "@boost//:scope_exit",
-        "@boost//:signals2",
-        "@boost//:system",
+        "@boost.chrono",
+        "@boost.filesystem",
+        "@boost.scope_exit",
+        "@boost.signals2",
```
[the remark this answers is not in the document]
Comment (hofbi) [reply ros/repositories/ros_comm.BUILD.bazel:136]: Now only "@boost.signals2", is missing

File: MODULE.bazel [hunk 2]
```diff
@@ -1,6 +1,18 @@
...
 
 bazel_dep(name = "bazel_skylib", version = "1.7.1")
+bazel_dep(name = "boost.any", version = "1.83.0")
+bazel_dep(name = "boost.chrono", version = "1.83.0")
+bazel_dep(name = "boost.date_time", version = "1.83.0")
+bazel_dep(name = "boost.filesystem", version = "1.83.0")
+bazel_dep(name = "boost.interprocess", version = "1.83.0")
+bazel_dep(name = "boost.regex", version = "1.83.0")
+bazel_dep(name = "boost.scope_exit", version = "1.83.0")
+bazel_dep(name = "boost.signals2", version = "1.83.0")
```
Comment (hofbi) [inline MODULE.bazel:11]: This is the only dep missing on BCR

Comment (wep21) [reply MODULE.bazel:11]: @hofbi `boost.signals2` is now available via BCR.

Comment (hofbi) [reply MODULE.bazel:12]: Just found that `boost_program_options` is another dep 🙈

Comment (wep21) [reply MODULE.bazel:12]: adding [here](https://github.com/bazelbuild/bazel-central-registry/pull/2950)

Comment (wep21) [reply MODULE.bazel:12]: need more deps https://github.com/hofbi/rules_ros/pull/1

File: MODULE.bazel [hunk 3]
```diff
@@ -1,6 +1,19 @@
 module(name = "rules_ros")
 
 bazel_dep(name = "bazel_skylib", version = "1.7.1")
+bazel_dep(name = "boost.any", version = "1.83.0")
+bazel_dep(name = "boost.chrono", version = "1.83.0")
+bazel_dep(name = "boost.date_time", version = "1.83.0")
+bazel_dep(name = "boost.filesystem", version = "1.83.0")
+bazel_dep(name = "boost.interprocess", version = "1.83.0")
```
Comment (hofbi) [inline MODULE.bazel:8]:
```suggestion
bazel_dep(name = "boost.interprocess", version = "1.83.0")
bazel_dep(name = "boost.math", version = "1.83.0")
```

File: ros/repositories/roscpp_core.BUILD.bazel [hunk 4]
```diff
@@ -37,9 +37,9 @@ cc_ros_library(
     visibility = ["//visibility:public"],
     deps = [
         ":cpp_common",
-        "@boost//:date_time",
-        "@boost//:system",
-        "@boost//:thread",
+        "@boost.date_time",
```
Comment (hofbi) [inline ros/repositories/roscpp_core.BUILD.bazel:40]:
```suggestion
        "@boost.date_time",
        "@boost.math",
```

File: MODULE.bazel (hunk 2 above)
Comment (hofbi) [reply MODULE.bazel:13]: Awesome, all deps on BCR and CI is green 🎉

**Comment by mvukov** [comment]
Nice work. We don't have a lot of tests in general. So, does `bazel build //...` work in examples?

**Comment by hofbi** [comment]
> Nice work. We don't have a lot of tests in general. So, does `bazel build //...` work in examples?

Yes it does. https://github.com/mvukov/rules_ros/actions/runs/11331836596/job/31512729687?pr=59 is running the examples.

**Comment by mvukov** [comment]
On CI we run `bazel test //...`. The test coverage can be better. Therefore, I am wondering whether `bazel build //...` still works.

File: MODULE.bazel [hunk 5]
```diff
@@ -1,6 +1,20 @@
 module(name = "rules_ros")
 
 bazel_dep(name = "bazel_skylib", version = "1.7.1")
+bazel_dep(name = "boost.any", version = "1.83.0")
+bazel_dep(name = "boost.chrono", version = "1.83.0")
+bazel_dep(name = "boost.date_time", version = "1.83.0")
+bazel_dep(name = "boost.filesystem", version = "1.83.0")
```
Comment (wep21) [inline MODULE.bazel:7]:
```suggestion
bazel_dep(name = "boost.filesystem", version = "1.83.0")
bazel_dep(name = "boost.format", version = "1.83.0")
```

File: examples/MODULE.bazel [hunk 6]
```diff
@@ -41,17 +41,3 @@ use_repo(
     "urdfdom",
     "urdfdom_headers",
 )
```
Comment (wep21) [inline examples/MODULE.bazel:43]:
```suggestion
)

bazel_dep(name = "boost.any", version = "1.83.0")
bazel_dep(name = "boost.smart_ptr", version = "1.83.0")
bazel_dep(name = "boost.thread", version = "1.83.0")

```

**Comment by hofbi** [comment]
What do you suggest to verify?

**Comment by mvukov** [comment]
I would suggest just run `cd examples && bazel build //...`. Nothing to be added to the CI at this moment.

**Comment by hofbi** [comment]
> I would suggest just run `cd examples && bazel build //...`. Nothing to be added to the CI at this moment.

I already did this an it works.

**Comment by mvukov** [comment]
Are suggestion from @wep21 going to be included?

**Comment by hofbi** [comment]
Oh, good point, though I already added them. Now they are there and just verified again that all samples build.

**Review by mvukov — APPROVED** [review]

**Pull request merged**